lib/gooddata/models/user_filters/user_filter_builder.rb

```diff
@@ -537,5 +539,17 @@ def self.normalize_filters(filters)
         end
       end
     end
+
+    # Removes MUFs from to_delete unless in user is in users_brick_input
+    def self.sanitize_filters_to_delete(to_delete, users_brick_input, project)
+      return to_delete unless users_brick_input && users_brick_input.any?
+      users = Hash[
+        project.users.map do |user|
+          [user.login, user.profile_url]
+        end
+      ]
+      user_profiles = users_brick_input.map { |user| users.find { |u| u[0] == user.login }[1] }
+      to_delete.reject { |uf, value| user_profiles.none? { |profile| profile == value.first.json[:related] } }
```

Comment:
Lint/UnusedBlockArgument: Unused block argument - `uf`. If it's necessary, use `_` or `_uf` as an argument name to indicate that it won't be used.